sipa/model/finance.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta, abstractmethod
+
+from flask_babel import gettext
+
+from sipa.model.fancy_property import ActiveProperty, Capabilities
+from sipa.units import format_money
+from sipa.model.misc import compare_all_attributes
+
+
+class BaseFinanceInformation(metaclass=ABCMeta):
+    """A Class providing finance information about a user.
+
+    This class bundles informations such as whether the current user
+    is obliged to pay, the current balance, history, and the time of
+    the last update.  The balance is being provided as a
+    FancyProperty, so it can be used as a row.
+
+    For subclassing, implement the respective abstract
+    methods/properties.
+    """
+
+    @property
+    def balance(self):
+        """The current balance as a
+        :py:class:`~sipa.model.fancy_property.ActiveProperty`
+
+        If :attr:`has_to_pay` is False, return an empty property with
+        the note that the user doesn't have to pay.  Else, return the
+        balance formatted and styled (red/green) as money and mark
+        this property editable.
+        """
+        if not self.has_to_pay:
+            return ActiveProperty('finance_balance', value=gettext("In Miete inbegriffen"),
```

Comment:
Now that I saw it on our testing instance, it should probably include the word “payment” as the row description reads “Finance balance” – A finance balance cannot be “included in [a] rental agreement”, only a payment can.